Source/WebCore/layout/integration/LayoutIntegrationCoverage.cpp

```diff
@@ -366,8 +366,8 @@ static OptionSet<AvoidanceReason> canUseForChild(const RenderBlockFlow& flow, co
             if (renderer.isFloating() && !renderer.parent()->style().isHorizontalWritingMode())
                 return false;
 #endif
-            if (renderer.isFloating() && flow.fragmentedFlowState() != RenderObject::NotInsideFragmentedFlow) {
-                // Floats inside fragmentation need integration support.
+            if (renderer.isFloating() && (flow.fragmentedFlowState() != RenderObject::NotInsideFragmentedFlow || !flow.style().marginTrim().isEmpty())) {
+                // Floats inside fragmentation and ones under margin-trim need integration support.
```

Comment:
nitpick: margin-trim does not need integration support. It needs IFC implementation (geometry adjustment that is).

Reply:
Changed